sdk/eventhubs/azure-messaging-eventhubs/src/main/java/com/azure/messaging/eventhubs/EventProcessorBuilder.java

```diff
@@ -117,8 +291,47 @@ public EventProcessorBuilder partitionProcessorFactory(Supplier<PartitionProcess
      * @return A new instance of {@link EventProcessor}.
      */
     public EventProcessor buildEventProcessor() {
+        Objects.requireNonNull(processEvent);
+        Objects.requireNonNull(eventProcessorStore);
+
         final TracerProvider tracerProvider = new TracerProvider(ServiceLoader.load(Tracer.class));
-        return new EventProcessor(eventHubAsyncClient, this.consumerGroup,
-            this.partitionProcessorFactory, EventPosition.earliest(), partitionManager, tracerProvider);
+        return new EventProcessor(eventHubClientBuilder, this.consumerGroup,
+            getPartitionProcessorSupplier(), EventPosition.earliest(), eventProcessorStore, tracerProvider);
+    }
+
+    private Supplier<PartitionProcessor> getPartitionProcessorSupplier() {
+        return () -> new PartitionProcessor() {
+            @Override
+            public Mono<Void> processEvent(PartitionEvent partitionEvent) {
+                return processEvent.apply(partitionEvent);
+            }
+
+            @Override
+            public Mono<Void> initialize(InitializationContext initializationContext) {
+                if (initializePartition != null) {
+                    return initializePartition.apply(initializationContext);
+                }
+                return super.initialize(initializationContext);
+            }
+
+            @Override
+            public void processError(ErrorContext errorContext) {
+                if (processError != null) {
+                    processError.apply(errorContext);
+                    return;
```

Comment:
nit: if else may make this easier to read.

Reply:
Done.